packages/select/src/common/predicate.ts

```diff
@@ -18,7 +18,7 @@
  * A custom predicate for returning an entirely new `items` array based on the provided query.
  * See usage sites in `IListItemsProps`.
  */
-export type ItemListPredicate<T> = (query: string, items: T[]) => T[];
+export type ItemListPredicate<T> = (query: string, items: readonly T[]) => T[];
```

Comment:
Changing the types of callback arguments may cause compile breaks for consumers, but should be easily fixed, since if they were mutating this argument here, it was probably causing problems